src/main/java/uz/uzfundit/Exercism/QueenAttackCalculator.java

```diff
@@ -0,0 +1,74 @@
+package uz.uzfundIT.Exercism;
+public class QueenAttackCalculator {
+    private final Queen whiteQueen;
+    private final Queen blackQueen;
+
+    QueenAttackCalculator(Queen whiteQueen, Queen blackQueen) {
+        if(whiteQueen == null || blackQueen == null) {
+            throw new IllegalArgumentException("You must supply valid positions for both Queens.");
+        } else if(whiteQueen.equals(blackQueen)) {
+            throw new IllegalArgumentException("Queens cannot occupy the same position.");
+        } else {
+            this.whiteQueen = whiteQueen;
+            this.blackQueen = blackQueen;
+        }
+    }
+
+    private boolean areQueensInSameRow(){
+        return whiteQueen.getRow() == blackQueen.getRow();
+    }
+
+    private boolean areQueensInSameColumn(){
+        return whiteQueen.getColumn() == blackQueen.getColumn();
+    }
+
+    private boolean areQueensInSameDiagonal(){
+        return (whiteQueen.getRow() - whiteQueen.getColumn()) == (blackQueen.getRow() - blackQueen.getColumn()) ||
+                (whiteQueen.getRow() + whiteQueen.getColumn()) == (blackQueen.getRow() + blackQueen.getColumn());
+    }
+
+    boolean canQueensAttackOneAnother() {
+        return areQueensInSameRow() || areQueensInSameColumn() || areQueensInSameDiagonal();
+    }
+}
+
+class Queen {
+
+    private final int row;
+    private final int column;
+
+    Queen(int row, int column) {
+        this.row = row;
+        this.column = column;
+        positionRow(row);
+        positionColumn(column);
+    }
```

Comment:
Although, here the effect is the same. You should validate first and then initialize fields